### PR TITLE
Rename "PartyTypes" and "CaseTypes" to "...TypeIDs" where relevant

### PR DESCRIPTION
**Pull request by nilueps**
As discussed with @ludydoo, this naming avoids the abiguity of the plural form which should be reserved for collections of the objects themselves.

### CODE REVIEW

File: api2/pkg/individuals/api.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func NewIndividual(ID string) *Individual {
 	return &Individual{
 		Party: &parties.Party{
 			ID: ID,
-			PartyTypes: []string{
+			PartyTypeIds: []string{
```
Comment (ludydoo) [inline api2/pkg/individuals/api.go:22]:
```suggestion
			PartyTypeIDs: []string{
```

File: api2/pkg/individuals/client.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func NewClient(basePath string) *Client {
 }
 
 type ListOptions struct {
-	PartyTypes []string `json:"partyTypes" bson:"partyTypes"`
+	PartyTypeIds []string `json:"partyTypeIds" bson:"partyTypeIds"`
```
Comment (ludydoo) [inline api2/pkg/individuals/client.go:23]:
```suggestion
	PartyTypeIDs []string `json:"partyTypeIds" bson:"partyTypeIds"`
```

File: api2/pkg/individuals/static.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ var FirstNameAttribute = attributes.Attribute{
 		},
 	},
 	IsPersonallyIdentifiableInfo: true,
-	PartyTypes: []string{
+	PartyTypeIds: []string{
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:22]:
```suggestion
	PartyTypeIDs: []string{
```

File: api2/pkg/individuals/rest.go [hunk 4]
```diff
@@ -49,7 +49,7 @@ func (h *Handler) List(w http.ResponseWriter, req *http.Request) {
 
 	ctx := req.Context()
 	listOptions := ListOptions{
-		PartyTypes: req.URL.Query()["partyTypes"],
+		PartyTypeIds: req.URL.Query()["partyTypeIds"],
```
Comment (ludydoo) [inline api2/pkg/individuals/rest.go:52]:
```suggestion
		PartyTypeIDs: req.URL.Query()["partyTypeIds"],
```

File: api2/pkg/individuals/static.go [hunk 5]
```diff
@@ -35,7 +35,7 @@ var LastNameAttribute = attributes.Attribute{
...
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:38]:
```suggestion
	PartyTypeIDs: []string{
```

File: api2/pkg/individuals/static.go [hunk 6]
```diff
@@ -51,7 +51,7 @@ var BirthDateAttribute = attributes.Attribute{
...
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:54]:
```suggestion
	PartyTypeIDs: []string{
```

File: api2/pkg/individuals/static.go [hunk 7]
```diff
@@ -65,26 +65,26 @@ var BuiltinIndividualAttributes = []attributes.Attribute{
 var mockIndividuals = []*Individual{
 	{
 		Party: &parties.Party{
-			ID:         "0bde06f0-5416-4514-9c5a-794a2cc2f1b7",
-			PartyTypes: []string{},
+			ID:           "0bde06f0-5416-4514-9c5a-794a2cc2f1b7",
+			PartyTypeIds: []string{},
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:69]:
```suggestion
			PartyTypeIDs: []string{},
```

File: api2/pkg/individuals/static.go [hunk 8]
```diff
@@ -65,26 +65,26 @@ var BuiltinIndividualAttributes = []attributes.Attribute{
...
 				FirstNameAttribute.ID: {"John"},
 				LastNameAttribute.ID:  {"Doe"},
 			},
 		},
 	}, {
 		Party: &parties.Party{
-			ID:         "ab7a1620-f34e-4811-8534-853167ed7944",
-			PartyTypes: []string{},
+			ID:           "ab7a1620-f34e-4811-8534-853167ed7944",
+			PartyTypeIds: []string{},
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:78]:
```suggestion
			PartyTypeIDs: []string{},
```

File: api2/pkg/individuals/static.go [hunk 9]
```diff
@@ -65,26 +65,26 @@ var BuiltinIndividualAttributes = []attributes.Attribute{
...
 				FirstNameAttribute.ID: {"Mary"},
 				LastNameAttribute.ID:  {"Poppins"},
 			},
 		},
 	}, {
 		Party: &parties.Party{
-			ID:         "40b30fb0-c392-4798-9400-bda3e5837867",
-			PartyTypes: []string{},
+			ID:           "40b30fb0-c392-4798-9400-bda3e5837867",
+			PartyTypeIds: []string{},
```
Comment (ludydoo) [inline api2/pkg/individuals/static.go:87]:
```suggestion
			PartyTypeIDs: []string{},
```

File: api2/pkg/individuals/store.go [hunk 10]
```diff
@@ -33,8 +33,8 @@ func (s *Store) Upsert(ctx context.Context, individual *Individual) error {
 		"id": individual.ID,
 	}, bson.M{
 		"$set": bson.M{
-			"attributes": individual.Attributes,
-			"partyTypes": individual.PartyTypes,
+			"attributes":   individual.Attributes,
+			"partyTypeIds": individual.PartyTypeIds,
```
Comment (ludydoo) [inline api2/pkg/individuals/store.go:37]:
```suggestion
			"partyTypeIds": individual.PartyTypeIDs,
```

File: api2/pkg/individuals/store.go [hunk 11]
```diff
@@ -60,9 +60,9 @@ func (s *Store) List(ctx context.Context, listOptions ListOptions) (*IndividualL
 
 	filter := bson.M{}
 
-	if len(listOptions.PartyTypes) > 0 {
-		filter["partyTypes"] = bson.M{
-			"$all": listOptions.PartyTypes,
+	if len(listOptions.PartyTypeIds) > 0 {
```
Comment (ludydoo) [inline api2/pkg/individuals/store.go:63]:
```suggestion
	if len(listOptions.PartyTypeIDs) > 0 {
```

File: api2/pkg/individuals/store.go [hunk 12]
```diff
@@ -60,9 +60,9 @@ func (s *Store) List(ctx context.Context, listOptions ListOptions) (*IndividualL
...
+		filter["partyTypeIds"] = bson.M{
+			"$all": listOptions.PartyTypeIds,
```
Comment (ludydoo) [inline api2/pkg/individuals/store.go:65]:
```suggestion
			"$all": listOptions.PartyTypeIDs,
```

File: api2/pkg/organizations/static.go [hunk 13]
```diff
@@ -19,15 +19,15 @@ var LegalNameAttribute = attributes.Attribute{
 			ShortFormulation: "Legal Name",
 		},
 	},
-	PartyTypes: []string{
+	PartyTypeIds: []string{
```
Comment (ludydoo) [inline api2/pkg/organizations/static.go:22]:
```suggestion
	PartyTypeIDs: []string{
```

File: api2/pkg/organizations/static.go [hunk 14]
```diff
@@ -19,15 +19,15 @@ var LegalNameAttribute = attributes.Attribute{
...
 		PartyType.ID,
 	},
 }
 
 var NRC = Organization{
 	Party: &parties.Party{
 		ID: "d0dc08b4-5e6b-461b-a444-09e02e69a8e1",
-		PartyTypes: []string{
+		PartyTypeIds: []string{
```
Comment (ludydoo) [inline api2/pkg/organizations/static.go:30]:
```suggestion
		PartyTypeIDs: []string{
```

File: api2/pkg/parties/attributes/api.go [hunk 15]
```diff
@@ -6,7 +6,7 @@ type Attribute struct {
 	ID                           string                 `json:"id" bson:"id"`
 	Name                         string                 `json:"name" bson:"name"`
 	ValueType                    expressions.ValueType  `json:"type" bson:"type"`
-	PartyTypes                   []string               `json:"partyTypes" bson:"partyTypes"`
+	PartyTypeIds                 []string               `json:"partyTypeIds" bson:"partyTypeIds"`
```
Comment (ludydoo) [inline api2/pkg/parties/attributes/api.go:9]:
```suggestion
	PartyTypeIDs                 []string               `json:"partyTypeIds" bson:"partyTypeIds"`
```

**Review by ludydoo — CHANGES_REQUESTED** [review]

**Comment by nilueps** [comment]
My bad, it should be fine now

**Review by ludydoo — APPROVED** [review]

**Pull request merged**